src/extension/appLauncher.ts

```diff
@@ -208,55 +213,84 @@ export class AppLauncher {
-                        return mobilePlatform.beforeStartPackager()
-                            .then(() => {
-                                generator.step("startPackager");
-                                return mobilePlatform.startPackager();
-                            })
-                            .then(() => {
-                                // We've seen that if we don't prewarm the bundle cache, the app fails on the first attempt to connect to the debugger logic
-                                // and the user needs to Reload JS manually. We prewarm it to prevent that issue
-                                generator.step("prewarmBundleCache");
-                                this.logger.info(localize("PrewarmingBundleCache", "Prewarming bundle cache. This may take a while ..."));
-                                return mobilePlatform.prewarmBundleCache();
-                            })
-                            .then(() => {
-                                generator.step("mobilePlatform.runApp").add("target", mobilePlatformOptions.target, false);
-                                this.logger.info(localize("BuildingAndRunningApplication", "Building and running application."));
-                                return mobilePlatform.runApp();
-                            })
-                            .then(() => {
-                                if (mobilePlatformOptions.isDirect || !mobilePlatformOptions.enableDebug) {
-                                    if (mobilePlatformOptions.isDirect && launchArgs.platform === "android") {
-                                        generator.step("mobilePlatform.enableDirectDebuggingMode");
-                                        if (mobilePlatformOptions.enableDebug) {
-                                            this.logger.info(localize("PrepareHermesDebugging", "Prepare Hermes debugging (experimental)"));
-                                        } else {
-                                            this.logger.info(localize("PrepareHermesLaunch", "Prepare Hermes launch (experimental)"));
-                                        }
+                        return new Promise((resolve) => {
+                            if (launchArgs.platform === "android" && mobilePlatform instanceof AndroidPlatform) {
+                                resolve(mobilePlatform.startEmulator(launchArgs.target));
+                            }
+                            else resolve(null);
+                        })
+                        .then((emulator: IAndroidEmulator | null) => {
+                            if (emulator) {
+                                let launchConfigIndex = this.launchScenariosManager.getFirstScenarioIndexByParams(launchArgs);
+                                const launchScenarios = this.launchScenariosManager.getLaunchScenarios();
+                                if (launchConfigIndex !== null && launchConfigIndex !== undefined && launchScenarios.configurations) {
```

Comment:
It cannot be undefined since the function returns number or null
```suggestion
                                if (launchConfigIndex && launchScenarios.configurations) {
```